### PR TITLE
Remove deprecated code, and deprecate some other code in Triangulation_2

### PR DESCRIPTION
**Pull request by afabri**
- `Triangulation_euclidean_traits_xy_3` got deprecated several releases ago.
- Deprecate `Triangulation_euclidean_traits_2` without replacement.

### CODE REVIEW

File: Triangulation_2/include/CGAL/Triangulation_euclidean_traits_2.h [hunk 1]
```diff
@@ -21,6 +21,9 @@
 #ifndef CGAL_TRIANGULATION_EUCLIDEAN_TRAITS_2_H
 #define CGAL_TRIANGULATION_EUCLIDEAN_TRAITS_2_H
```
Comment (sloriot) [inline Triangulation_2/include/CGAL/Triangulation_euclidean_traits_2.h]: You removed the other deprecated header but not this one. Any reason for that choice?

Comment (sloriot) [inline Triangulation_2/include/CGAL/Triangulation_euclidean_traits_2.h]: Sorry you added a deprecation notice. Filename is not correct btw (uppercase t)

File: Triangulation_2/test/Triangulation_2/test_delaunay_triangulation_proj.cpp [hunk 2]
```diff
@@ -3,22 +3,22 @@
 #define CGAL_NO_DEPRECATION_WARNINGS 1
```
Comment (sloriot) [inline Triangulation_2/test/Triangulation_2/test_delaunay_triangulation_proj.cpp]: this "#define" is useless now

**Comment by afabri** [comment]
fixed

File: Triangulation_2/test/Triangulation_2/test_delaunay_triangulation_proj.cpp [hunk 3]
```diff
@@ -1,24 +1,23 @@
 // This file tests the deprecated headers
 // Triangulation_euclidean_traits_.._3 on purpose.
-#define CGAL_NO_DEPRECATION_WARNINGS 1
```
Comment (lrineau) [inline Triangulation_2/test/Triangulation_2/test_delaunay_triangulation_proj.cpp]: The comment just above that line can be removed too.

**Comment by lrineau** [comment]
It seems that this PR introduces breaking changes. It should be announced in the `changes.html`.

**Comment by sloriot** [comment]
cc @afabri

**Pull request merged**